### DISCUSSION TITLE
Friends unable to join

### DISCUSSION THREAD

**Discussion by Kuggle01** [Pryaxis/TShock#2230] [General]
So I started up my Tshock server just fine, I was able to join using the IP 127.0.0.1 and port 7777 but none of my friends are able to join no matter what, I've tried having a password and not having a password, I've tried having auto-port fowarding on and off, but no matter what I do nobody but me can connect to the server, one of my friends is telling me that I'm using the wrong IP but I do not know what IP I should be using or how to even change the IP that the server runs. thank you in advance and please get back to me at the soonest conveyance. I followed a guide that has not been updated for 2 years tho so it's likely that the steps that I followed in that guide were outdated.

**Comment by Thunder1723** [comment]
While this post is 14 days old if you're still having trouble you can try two things.  Learn how to port forward or find your public IP.  The latter is much easier and I would suggest doing that first.  Going to the website  https://www.whatismyip.com/  will tell you your public IP.  I personally have yet to use port forwarding because finding my public IP took 5 seconds.  Using that should work and it should stay the same.  I understand it is difficult using outdated guides, most of the threads I used to make my raspberry pi run were always 5+ years old.  I'm not knowledgeable enough to write a guide but that's what worked for me.

**Comment by EMREOYUN** [comment]
> So I started up my Tshock server just fine, I was able to join using the IP 127.0.0.1 and port 7777 but none of my friends are able to join no matter what, I've tried having a password and not having a password, I've tried having auto-port fowarding on and off, but no matter what I do nobody but me can connect to the server, one of my friends is telling me that I'm using the wrong IP but I do not know what IP I should be using or how to even change the IP that the server runs. thank you in advance and please get back to me at the soonest conveyance. I followed a guide that has not been updated for 2 years tho so it's likely that the steps that I followed in that guide were outdated.

Setting your IP to 127.0.0.1 directly closes the port to external connections and it makes the port can be reachable from your device only. Please try setting your IP to 0.0.0.0 on config and try again.
